### PR TITLE
Fixing "AzCopy from S3 to Blob fails with 401 code after update"

### PR DESCRIPTION
**Pull request by siminsavani-msft**
Fix for https://github.com/Azure/azure-storage-azcopy/issues/1551

### CODE REVIEW

File: cmd/credentialUtil.go [hunk 1]
```diff
@@ -562,8 +562,6 @@ func GetCredentialInfoForLocation(ctx context.Context, location common.Location,
 func getCredentialType(ctx context.Context, raw rawFromToInfo, cpkOptions common.CpkOptions) (credType common.CredentialType, err error) {
 
 	switch {
-	case raw.fromTo == common.EFromTo.S3Blob(): // S3 Bucket
-		credType, _, err = getCredentialTypeForLocation(ctx, raw.fromTo.From(), raw.source, raw.sourceSAS, true, common.CpkOptions{})
```
Comment (mohsha-msft) [inline cmd/credentialUtil.go:566]: Was this code added in 10.12?
If not, was this scenario working in 10.11?

Comment (siminsavani-msft) [reply cmd/credentialUtil.go:566]: Yes, it was added in 10.12! 

I also tested 10.11 and this scenario was working.

**Review by zezha-msft — COMMENTED** [review]
So this was a regression and we don't have a test, can the e2e package accommodate such a test right now?

**Comment by siminsavani-msft** [comment]
> So this was a regression and we don't have a test, can the e2e package accommodate such a test right now?

I will look into adding tests for this scenario!

**Comment by mohsha-msft** [comment]
> So this was a regression and we don't have a test, can the e2e package accommodate such a test right now?

We currently don't have support for both S3 and GCP in our E2E test framework. Shall we pick it up for 10.14 or next cleanup week? @zezha-msft

**Pull request merged**